apps/api/src/versioncontrol/test/representative.spec.ts

```diff
@@ -35,84 +39,118 @@ describe("Representative", () => {
+      const parsed = {
+        ...resource,
         contents: {
-          schema: {title: "hi"}
+          ...resource.contents,
+          package: JSON.parse(resource.contents.package),
+          schema: YAML.parse(resource.contents.schema)
         }
-      });
-    });
-
-    it("should read content of multiple files", async () => {
-      await representative.write(
-        "module1",
-        "id1",
-        "package",
-        {dependencies: {dep1: "1.1"}},
-        "json"
-      );
-      await representative.write("module1", "id1", "schema", {title: "hi"}, "yaml");
-      await representative.write("module1", "id1", "index", "console.log(123)", "js");
-      await representative.write("module1", "id1", "env", {APIKEY: "SECRET"}, "env");
-
-      const resource = await representative.readResource("module1", "id1");
-      expect(resource).toEqual({
+      };
+      expect(parsed).toEqual({
         _id: "id1",
         contents: {
           package: {dependencies: {dep1: "1.1"}},
           schema: {title: "hi"},
-          index: "console.log(123)",
-          env: {
-            APIKEY: "SECRET"
-          }
+          index: "console.log(123)"
         }
       });
     });
+  });
 
-    it("should read all contents of module", async () => {
-      await representative.write(
-        "module1",
-        "id1",
-        "package",
-        {dependencies: {dep1: "1.1"}},
-        "json"
-      );
-      await representative.write("module1", "id1", "schema", {title: "hi"}, "yaml");
-      await representative.write("module1", "id1", "index", "console.log(123)", "js");
+  describe("watch", () => {
+    let subscribtion: Subscription;
+
+    it("should track insert change type", done => {
+      subscribtion = representative.watch("module1", ["schema.yaml"]).subscribe({
+        next: change => {
+          const parsedContent = YAML.parse(change.resource.content);
+          change.resource = {...change.resource, content: parsedContent};
+
+          expect(change).toEqual({
+            resourceType: ResourceType.REPRESENTATIVE,
+            changeType: ChangeTypes.INSERT,
+            resource: {_id: "id1", content: {title: "hi"}}
+          });
+          subscribtion.unsubscribe();
+          done();
+        }
+      });
+
+      const stringified = YAML.stringify({title: "hi"});
+      representative.write("module1", "id1", "schema", stringified, "yaml");
+    });
 
-      const contents = await representative.read("module1", () => true);
-      expect(contents).toEqual([
-        {
-          _id: "id1",
-          contents: {
-            package: {dependencies: {dep1: "1.1"}},
-            schema: {title: "hi"},
-            index: "console.log(123)"
-          }
+    it("should track update change type", done => {
+      subscribtion = representative.watch("module1", ["schema.yaml"]).subscribe({
+        next: change => {
+          if (change.changeType == ChangeTypes.INSERT) return;
```

Comment:
Could you use RxJS operators to skip the first message instead of this if check?

Reply:
Because if the update change comes before the insert, tests will be passed, which should be failed.